pkg/bff/admin.go

```diff
@@ -354,3 +354,27 @@ func (s *Server) Attention(c *gin.Context) {
 		})
 	}
 }
+
+// RegistrationStatus returns the registration status for both testnet and mainnet for
+// the user.
+func (s *Server) RegistrationStatus(c *gin.Context) {
+	var err error
+
+	// Retrieve the organization from the claims
+	// NOTE: This method handles the error logging and response.
+	var org *records.Organization
+	if org, err = s.OrganizationFromClaims(c); err != nil {
+		return
+	}
+
+	// Build the response
+	// TODO: We should be querying the VASP record instead to allow for re-registration
+	out := &api.RegistrationStatus{}
+	if org.Testnet != nil && org.Testnet.Submitted != "" {
+		out.TestNetSubmitted = org.Testnet.Submitted
+	}
+	if org.Mainnet != nil && org.Mainnet.Submitted != "" {
+		out.MainNetSubmitted = org.Mainnet.Submitted
+	}
```

Comment:
Currently it doesn't look like the frontend allows a path to re-registration, because the bff doesn't know which state the VASP is in. In order to make this possible, I think we need to check the status of the VASP in GDS rather than using the submitted field on the organization, since those fields don't get reset. In order to do that, we really need to have a bff-specific API for the bff to communicate with the Admin API, which is a larger effort.

Reply:
Hmm, that's a good point -- does that mean re-registration currently means wiping the state stored by the front end and starting over?

Reply:
Yes I think we would have to update the organization record in the database. This PR just allows the frontend to know if the form has already been submitted before allowing the user to submit a registration (they currently get an error when clicking the submit button).